addons/mail/models/mail_thread.py

```diff
@@ -2027,7 +2027,7 @@ def message_post(self, body='', subject=None,
             'parent_id': parent_id,
             'subtype_id': subtype_id,
             'partner_ids': [(4, pid) for pid in partner_ids],
-            'channel_ids': kwargs.get('channel_ids', []),
+            'channel_ids': [(4, cid) for cid in kwargs.get('channel_ids', [])],
```

Comment:
Actually most of message_post arguments should be like a new message creation values. Therefore either you give a valid orm command to message_post in JS, either you process them before to correctly support either a list of ids, either a list of commands. Code here is not correct as you break valid behavior for methods already giving a valid ORM command for channel_ids .